python_modules/dagster/dagster/_core/definitions/metadata/metadata_value.py

```diff
@@ -609,7 +613,31 @@ def value(self) -> Optional[str]:
         return self.path_inner
 
 
-@whitelist_for_serdes(storage_name="JsonMetadataEntryData")
+class JsonDataFieldSerializer(FieldSerializer):
+    def pack(
+        self,
+        mapping: JsonSerializableValue,
+        whitelist_map: WhitelistMap,
+        descent_path: str,
+    ) -> JsonSerializableValue:
+        # return the json serializable data field as is
+        return mapping
+
+    def unpack(
+        self,
+        unpacked_value: JsonSerializableValue,
+        whitelist_map: WhitelistMap,
+        context: UnpackContext,
+    ) -> PackableValue:
+        # erase any serdes objects that were stored here in earlier versions and
+        # return plain json serializable data
+        return pack_value(unpacked_value, whitelist_map=whitelist_map)
```

Comment:
because we deserialize bottoms-up the only way we can remove any serdes objects that showed up is to recurse back down the tree searching for them. Any time we have lots of json metadata this will end up being costly

Reply:
ran benchmarks with some sample large serdes objects, no measurable regression